keras_nlp/models/whisper/whisper_backbone.py

```diff
@@ -0,0 +1,262 @@
+    model with any number of layers, heads, and embedding dimensions. To load
+    preset architectures and weights, use the `from_preset` constructor.
+
+    Disclaimer: Pre-trained models are provided on an "as is" basis, without
+    warranties or conditions of any kind. The underlying model is provided by a
+    third party and subject to a separate license, available
+    [here](https://github.com/openai/whisper).
+
+    Args:
+        vocabulary_size: int. The size of the token vocabulary.
+        num_layers: int. The number of transformer encoder layers and
+            transformer decoder layers.
+        num_heads: int. The number of attention heads for each transformer.
+            The hidden size must be divisible by the number of attention heads.
+        hidden_dim: int. The size of the transformer encoding and pooler layers.
+        intermediate_dim: int. The output dimension of the first Dense layer in
+            a two-layer feedforward network for each transformer.
+        num_mels: int. The number of mel-frequency filters. For now, only 80
+            works.
+        dropout: float. Dropout probability for the Transformer encoder.
+        max_source_sequence_length: int. The maximum sequence length that the
+            audio encoder can consume. Note that this is not the sequence length
+            of the encoder input since the input is passed through a couple of
+            convolutional layers, the second of which has a stride of 2. Hence,
+            `max_source_sequence_length` is generally half of what the maximum
+            encoder input length is.
+        max_target_sequence_length: int. The maximum sequence length that the
+            text decoder can consume.
+    """
+
+    def __init__(
+        self,
+        vocabulary_size,
+        num_layers,
+        num_heads,
+        hidden_dim,
+        intermediate_dim,
+        num_mels=80,
+        dropout=0.0,
+        max_source_sequence_length=1500,
+        max_target_sequence_length=448,
+        **kwargs,
+    ):
+        # Encoder inputs. Note that the encoder does not have a padding mask:
+        # https://github.com/openai/whisper/blob/v20230124/whisper/model.py#L132.
+        # This is because we have two convolutional layers in the encoder, post
+        # which the sequence dimension reduces to half.
+        encoder_feature_input = keras.Input(
+            shape=(None, num_mels), dtype="float32", name="encoder_features"
+        )
+
+        # Decoder inputs.
+        decoder_token_id_input = keras.Input(
+            shape=(None,), dtype="int32", name="decoder_token_ids"
+        )
+        decoder_padding_mask = keras.Input(
+            shape=(None,), dtype="int32", name="decoder_padding_mask"
+        )
+
+        # ====== Encoder ======
+
+        # Embed the input features. This consists of two 1D convolutional
+        # layers.
+        # For the first conv. layer, We use `padding="same"` here since that
```

Comment:
For the first conv. layer, We -> For the first layer, we

(you already mention convolution above the and the punctuation and capitalization is a little confusing)